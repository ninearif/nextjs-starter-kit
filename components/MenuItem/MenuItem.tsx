import { ReactNode } from 'react';

type Props = {
  icon: ReactNode;
  text: string;
  href: string;
};

const MenuItem = ({ icon, text, href }: Props) => {
  return (
    <a href={href}>
      <div className="btn btn-ghost gap-2">
        {icon}
        <b>{text}</b>
      </div>
    </a>
  );
};

export default MenuItem;
