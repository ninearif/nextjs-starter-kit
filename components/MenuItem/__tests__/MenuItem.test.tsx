import { render, screen } from '@testing-library/react';

import MenuItem from '../MenuItem';

describe('MenuItem component test', () => {
  it('MenuItem is rendered as expected.', () => {
    render(<MenuItem href="#test" text="test menu" icon />);

    expect(screen.getByText('test menu')).toBeInTheDocument();
    // expect(screen.getByRole("heading")).toHaveTextContent(/MenuItem/);
    // expect(screen.getByRole("button", { name: "submit" })).toBeDisabled();
    // expect(screen.getByRole("img")).toBeInTheDocument();
    // some cheat sheet can be found here:
    // https://testing-library.com/docs/react-testing-library/cheatsheet/
  });
});

describe('MenuItem component snapshot test', () => {
  it('render MenuItem unchanged', () => {
    const { container } = render(<MenuItem />);
    expect(container).toMatchSnapshot();
  });
});
