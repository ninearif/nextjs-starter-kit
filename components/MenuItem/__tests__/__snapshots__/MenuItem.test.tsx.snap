// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component snapshot test render MenuItem unchanged 1`] = `
<div>
  <a>
    <div
      class="btn btn-ghost gap-2"
    >
      <b />
    </div>
  </a>
</div>
`;
